x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -140,16 +142,33 @@ public void authenticate(String action, TransportMessage message, User fallbackU
     }
 
     /**
-     * Authenticates the username and password that are provided as parameters. This will not look
-     * at the values in the ThreadContext for Authentication.
+     * Authenticates the user based on the contents of the token that is provided as parameter.If {@code preferredRealm} can't
+     * authenticate the token, no other realms will be attempted.
+     * This will not look at the values in the ThreadContext for Authentication.
      *
      * @param action  The action of the message
      * @param message The message that resulted in this authenticate call
      * @param token   The token (credentials) to be authenticated
      */
     public void authenticate(String action, TransportMessage message,
                              AuthenticationToken token, ActionListener<Authentication> listener) {
-        new Authenticator(action, message, null, listener).authenticateToken(token);
+        new Authenticator(action, message, null, listener).authenticateToken(token, null);
+    }
+
+    /**
+     * Authenticates the user based on the contents of the token that is provided as parameter only at the realm that is indicated with
+     * {@code preferredRealm}. If {@code preferredRealm} can't authenticate the token, no other realms will be attempted.
+     * This will not look at the values in the ThreadContext for Authentication.
+     *
+     * @param action  The action of the message
+     * @param message The message that resulted in this authenticate call
+     * @param token   The token (credentials) to be authenticated
+     * @param preferredRealm The realm that should be used to authenticate the token
+     */
+    public void authenticate(String action, TransportMessage message,
```

Comment:
Opted to overloading public methods and just adding Nullable parameters to private ones.